### DISCUSSION TITLE
Is webgl being used to render the bars

### DISCUSSION THREAD

**Discussion by sualehasif** [tradingview/lightweight-charts#1192] [Q&A] [answered]
It seems that on an M1, the maximum number of bars that chrome can render smoothly is ~40k. It seems that the CPU is being used to render these? Am I correct. Is it possible to use a gpu render?

**Comment by SlicedSilver** [comment, marked as answer]
The library isn't using webGL, but rather the 2D rendering context of a canvas element. There isn't an option to switch to webGL with the library.

The maximum number of bars visible at a time is determined by the size of the screen since a bar can't become smaller than a pixel wide. If you are rendering multiple series at a time then it can obviously render more bars on the screen. There will always be a point where the cpu usage will reach the limit of the machine. Perhaps if you are willing to share your example then we could dig a bit deeper into the performance and see if there is anything we can improve.
